cmake/README.md

```diff
@@ -1,60 +1,243 @@
+    X:\>
+
+Where *X* is your working drive letter. And change directory to your working folder:
+
+    X:\>cd Path\to
+    X:\Path\to>
+
+Where *Path\to* is path to your real working folder.
+
+Make folder where builded protobuf will be installed:
+
+    X:\Path\to>mkdir install
+
+If *cmake* coomand is not avaliable from *Command Promt*, add it in system *PATH* variable:
+
+    X:\Path\to>set PATH=%PATH%;C:\Program Files (x86)\CMake\bin
+
+If *git* coomand is not avaliable from *Command Promt*, add it in system *PATH* variable:
+
+    X:\Path\to>set PATH=%PATH%;C:\Program Files\Git\cmd
+
+Good. Now you are ready to continue.
+
+Getting Sources
+===============
+
+You can get the latest stable source packages from the
+[releases](https://github.com/google/protobuf/releases) page.
+Or you can type:
+
+     X:\Path\to> git clone -b release_tag https://github.com/google/protobuf.git
+
+Where *release_tag* is a real repository tag or branch name as a *master*
+if you want to get the latest code.
+
+Go to the project folder:
+
+     X:\Path\to>cd protobuf
+     X:\Path\to\protobuf>
+
+Check whether a *gmock* directory exists in the upper level directory. If you
+checkout the code from GitHub via `git clone`, this *gmock* directory won't
+exist and you won't be able to build protobuf unit-tests.
+
+These source packages from *releases* page contains more stable versions of protobuf
+and already have the *gmock* directory included.
+
+If even so the *gmock* folder does not exist, you can download it as follows:
+
+     X:\Path\to\protobuf>git clone -b release-1.7.0 https://github.com/google/googlemock.git gmock
+
+Then go to *gmock* folder and downdload gtest:
+
+     X:\Path\to\protobuf>cd gmock
+     X:\Path\to\protobuf\gmock>git clone -b release-1.7.0 https://github.com/google/googletest.git gtest
+
+If you absolutely don't want to build and run protobuf unit-tests, skip
+this steps and use protobuf at your own risk.
+
+Now go to *cmake* folder in protobuf sources:
+
+     X:\Path\to\protobuf\gmock>cd ..\cmake
+     X:\Path\to\protobuf\cmake>
+
+Good. Now you are ready to *CMake* configuration.
+
+CMake Configuration
+===================
+
+The *CMake* provides a lot of different
+[generators](http://www.cmake.org/cmake/help/latest/manual/cmake-generators.7.html)
+for various native build system.
+We are interested a
+[Makefile](http://www.cmake.org/cmake/help/latest/manual/cmake-generators.7.html#makefile-generators)
+and a
+[Visual Studio](http://www.cmake.org/cmake/help/latest/manual/cmake-generators.7.html#visual-studio-generators)
+generators.
+
+We will use shadow building to separate the temporary files from the protobuf source code.
+
+Make temporary *build* folder and change directory to:
+
+     X:\Path\to\protobuf\cmake>mkdir build & cd build
+     X:\Path\to\protobuf\cmake\build>
+
+The *Makefile* generator can build the project in only one configuration, so you need to build
```

Comment:
Can we just describe how to build with Visual Studio IDE and remove the nmake stuff? The current description might confuse users and gives them a wrong impression that they need to follow all these steps.